tests/test_views.py

```diff
@@ -0,0 +1,174 @@
+import os
+import json
+import csv
+import datetime
+import psycopg2
+import pytest
+from dotenv import load_dotenv
+
+load_dotenv()
+DATABASE_URL = os.getenv('DATABASE_URL',
+    'postgres://app_user:secret@localhost:5432/app_db'
+)
+
+# Path helper
+DATA_DIR = os.path.join(os.path.dirname(__file__), "data")
+
+def insert_mock_data(cursor, data):
+    cursor.execute("DELETE FROM api.events;")
+    for record in data:
+        cursor.execute(
+            "INSERT INTO api.events (event) VALUES (%s);",
+            (json.dumps(record),)
+        )
+
+def query_events_view(cursor):
+    cursor.execute("""
+        SELECT schema_version, occurred_at, user_id, event_type
+          FROM api.events_view
+      ORDER BY occurred_at;
+    """)
+    return cursor.fetchall()
+
+def load_expected(path):
+    """Return list of tuples (schema_version, occurred_at_str, user_id, event_type)."""
+    with open(path, newline='') as f:
+        reader = csv.DictReader(f)
+        rows = []
+        for row in reader:
+            rows.append((
+                row['schema_version'],
+                row['occurred_at'],
+                row['user_id'],
+                row['event_type'],
+            ))
+    return rows
+
+def format_timestamp(dt):
+    """Render a datetime as 'YYYY-MM-DD HH:MM:SS' (no tz offset)."""
+    return dt.strftime('%Y-%m-%d %H:%M:%S')
+
+def normalize_results(rows):
+    """
+    Given an iterable of (schema, datetime, user_id, event_type),
+    return a list of tuples where the datetime is formatted as
+    'YYYY-MM-DD HH:MM:SS' plus '+00' only for 'ltt' rows.
+    """
+    out = []
+    for schema, dt, user, ev in rows:
+        ts = dt.strftime('%Y-%m-%d %H:%M:%S')
+        if schema != 'legacy':
+            ts += '+00'
+        out.append((schema, ts, user, ev))
+    return out
+
+def run_test(mock_data, expected_rows):
+    """Wipe, insert mock_data, run view, normalize & assert against expected_rows."""
+    with psycopg2.connect(DATABASE_URL) as conn, conn.cursor() as cur:
+        insert_mock_data(cur, mock_data)
+        conn.commit()
+        raw = query_events_view(cur)
+    actual = normalize_results(raw)
+    assert actual == expected_rows
```

Comment:
this might make more sense as a fixture for the psql connection, and the asserts listed in each individual test. wrapping every test in the same function makes it harder to see what's happening in a failing test